website/source/docs/upgrading.html.md

```diff
@@ -36,6 +36,18 @@ Consul is A, and version B is released.
    by running `consul members` to make sure all members have the latest
    build and highest protocol version.
 
+## Upgrade from Version 1.0.6 to higher
+
+In version 1.0.7 and higher, when requesting a specific service (/health or
+/catalog endpoints), the X-Consul-Index returned is now the index at which the
+service has been modified, not the global Raft Index of all services.
+
+Thus, if several versions of Consul are pre 1.0.7 and post 1.0.7, (ie: during an
+upgrade) it is possible to have a lower X-Consul-Index returned than the previous
+X-Consul-Index issued for this service.
+
+It should not be an issue unless library code does issue an error if it expects
```

Comment:
Rewrite this as: 
```
This may cause unexpected results in features like Consul watches, and in tools like consul-template, both of which rely on non blocking queries.
```

Reply:
+1 on the change.

We could maybe also add

> In most cases restarting the watch with the new index should be fine however care should be taken with tools that persist indexes anywhere or explicitly rely on them being monotonically increasing.

Reply:
+1 to all of @banks suggestions

Reply:
@preetapan and I double checked both `consul watch` and `consul-template` and they should both handle this.

So maybe we can replace the whole sentence (instead of our previous suggestions) with:

> Consul’s `watch` feature and `consul-template` should gracefully handle this case. Other tools relying on blocking service or health queries are also likely to work; some may require a restart. It is possible external tools could break and either stop working or continually re-request data without blocking if they have assumed indexes can never decrease or be reset and/or persist index values. Please test any blocking query integrations in a controlled environment before proceeding.

Reply:
Ok, DONE in next patch